### PR TITLE
Fix version after which org.graalvm.nativeimage needs to be exported

### LINKED ISSUE #36053: WARNING: Unknown module: org.graalvm.nativeimage specified to --add-exports when using Mandrel 23.0.1
### Describe the bug

Hi. I see `WARNING: Unknown module: org.graalvm.nativeimage specified to --add-exports` when using Mandrel 23.0.1 and Quarkus main

This must be related to https://github.com/quarkusio/quarkus/pull/35377

```
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Running Quarkus native-image plugin on native-image 17.0.8 2023-07-18
OpenJDK Runtime Environment Mandrel-23.0.1.2-Final (build 17.0.8+7)
OpenJDK 64-Bit Server VM Mandrel-23.0.1.2-Final (build 17.0.8+7, mixed mode)
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildRunner] /root/.sdkman/candidates/java/23.0.1.2.r17-mandrel/bin/native-image -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Dlogging.initial-configurator.min-level=500 -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=3 -J-Duser.language=en -J-Duser.country=US -J-Dfile.encoding=UTF-8 --features=io.quarkus.runner.Feature,io.quarkus.runtime.graal.DisableLoggingFeature -J--add-exports=java.security.jgss/sun.security.krb5=ALL-UNNAMED -J--add-opens=java.base/java.text=ALL-UNNAMED -J--add-opens=java.base/java.io=ALL-UNNAMED -J--add-opens=java.base/java.lang.invoke=ALL-UNNAMED -J--add-opens=java.base/java.util=ALL-UNNAMED -H:BuildOutputJSONFile=quarkus-runner-build-output-stats.json -H:Log=registerResource: -H:IncludeResources=privateKey.pem -H:+AllowFoldMethods -J-Djava.awt.headless=true --no-fallback --link-at-build-time -H:+ReportExceptionStackTraces -H:-AddAllCharsets --enable-url-protocols=http,https -H:NativeLinkerOption=-no-pie --enable-monitoring=heapdump -H:-UseServiceLoaderFeature -H:+StackTrace -J--add-exports=org.graalvm.sdk/org.graalvm.nativeimage.impl=ALL-UNNAMED -J--add-exports=org.graalvm.nativeimage/org.graalvm.nativeimage.impl=ALL-UNNAMED --exclude-config io\.netty\.netty-codec /META-INF/native-image/io\.netty/netty-codec/generated/handlers/reflect-config\.json --exclude-config io\.netty\.netty-handler /META-INF/native-image/io\.netty/netty-handler/generated/handlers/reflect-config\.json quarkus-runner -jar quarkus-runner.jar
WARNING: Unknown module: org.graalvm.nativeimage specified to --add-exports
```

### Expected behavior

No warning shown when using Mandrel 23.0.1

### Actual behavior

WARNING: Unknown module: org.graalvm.nativeimage specified to --add-exports when using Mandrel 23.0.1

### How to Reproduce?

 - `sdk use java 23.0.1.2.r17-mandrel`
 - `mvn clean package -Dnative` on any Quarkus project

### Output of `uname -a` or `ver`

Linux

### Output of `java -version`

Java 17

### GraalVM version (if different from Java)

Mandrel 23.0.1

### Quarkus version or git rev

76348f3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by zakkak**
The goal is to export the module after 23.0.x not after `23.0.0`, i.e., when using `23.0.1` we should not export it.

Wrong version introduced in https://github.com/quarkusio/quarkus/pull/35377

Closes: https://github.com/quarkusio/quarkus/pull/36053

### CODE REVIEW

**Review by jerboaa — APPROVED** [review]
Looks good. We had to do a similar change in mandrel integration tests: https://github.com/Karm/mandrel-integration-tests/pull/200

File: core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java:56]: Wouldn't it be easier if the version here was the version from where we need to add this, rather than the version before?

I.e. you would use `23.1.0` here.

I will merge this but I wonder if we should change the behavior in the future.

**Review by gsmet — APPROVED** [review]

File: core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java
Comment (zakkak) [reply core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java:56]: Yes, that makes sense. I can prepare a PR for it.

Comment (zakkak) [reply core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java:56]: See https://github.com/quarkusio/quarkus/pull/36063

**Comment by gsmet** [comment]
OK, let's not merge this right away then and discuss things in https://github.com/quarkusio/quarkus/pull/36063 .

**Pull request closed without merging**
